server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -4370,6 +4377,22 @@ public void close() throws IOException {
         onSettingsChanged();
     }
 
+    public void syncTranslogFilesFromRemoteTranslog() throws IOException {
+        TranslogFactory translogFactory = translogFactorySupplier.apply(indexSettings, shardRouting);
+        assert translogFactory instanceof RemoteBlobStoreInternalTranslogFactory;
+        Repository repository = ((RemoteBlobStoreInternalTranslogFactory) translogFactory).getRepository();
+        assert repository instanceof BlobStoreRepository : "repository should be instance of BlobStoreRepository";
+        BlobStoreRepository blobStoreRepository = (BlobStoreRepository) repository;
+        FileTransferTracker fileTransferTracker = new FileTransferTracker(shardId);
+        TranslogTransferManager translogTransferManager = RemoteFsTranslog.buildTranslogTransferManager(
+            blobStoreRepository,
+            getThreadPool(),
+            shardId,
+            fileTransferTracker
+        );
+        RemoteFsTranslog.download(translogTransferManager, shardPath().resolveTranslog());
+    }
```

Comment:
This entire logic should sit in RemoteFsTranslog and IndexShard should ever interact with a common `Translog` interface. Otherwise every consumer will need to be aware of the underlying translog which will make the interfaces too tightly coupled.

Reply:
Makes sense, will make the changes accordingly.

Reply:
Tracking it here: https://github.com/opensearch-project/OpenSearch/issues/5679